CommonTools/ParticleFlow/plugins/ChargedHadronIsolationProducer.cc

```diff
@@ -0,0 +1,86 @@
+/*
+ * ChargedHadronIsolationProducer
+ *
+ * Author: Andreas Hinzmann
+ *
+ * Associates PF isolation flag to charged hadron candidates
+ *
+ */
+
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidateFwd.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlock.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElement.h"
+
+class ChargedHadronIsolationProducer : public edm::stream::EDProducer<> 
+{
+public:
+
+  explicit ChargedHadronIsolationProducer(const edm::ParameterSet& cfg);
+  ~ChargedHadronIsolationProducer() {}
+  void produce(edm::Event& evt, const edm::EventSetup& es);
+
+private:
+  // input collection
+  edm::InputTag srcCandidates_;
+  edm::EDGetTokenT<edm::View<reco::PFCandidate> > Candidates_token;
+  double minTrackPt_;
+  double minRawCaloEnergy_;
+
+};
+
+ChargedHadronIsolationProducer::ChargedHadronIsolationProducer(const edm::ParameterSet& cfg)
+{
+  srcCandidates_ = cfg.getParameter<edm::InputTag>("src");
+  Candidates_token = consumes<edm::View<reco::PFCandidate> >(srcCandidates_);
+  minTrackPt_ = ( cfg.exists("minTrackPt") ) ? cfg.getParameter<double>("minTrackPt") : -1.0;
+  minRawCaloEnergy_ = ( cfg.exists("minRawCaloEnergy") ) ? cfg.getParameter<double>("minRawCaloEnergy") : -1.0;
```

Comment:
all of these "if (exists" conditionals should be removed and replaced with defaults defined in fillDescriptions
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp